### DISCUSSION TITLE
`npm run build` fails on aws only

### DISCUSSION THREAD

**Discussion by ivanakimov** [vercel/next.js#28986] [Help]
### What version of Next.js are you using?

11.1.2

### What version of Node.js are you using?

node:alpine

### What browser are you using?

any browser

### What operating system are you using?

node:alpine on aws

### How are you deploying your application?

next build

### Describe the Bug

My `npm run build` works locally.
My `Dockerfile` successfully builds locally as well.

But the same Dockerfile refuses to work on AWS. This is what I get:

```
Step 15/18 : RUN npm run build
 ---> Running in 95855ddd64d0

> x@0.1.0 build
> next build

info  - Using webpack 5. Reason: Enabled by default https://nextjs.org/docs/messages/webpack5
Attention: Next.js now collects completely anonymous telemetry regarding usage.
This information is used to shape Next.js' roadmap and prioritize features.
You can learn more, including how to opt-out if you'd not like to participate in this anonymous program, by visiting the following URL:
https://nextjs.org/telemetry

info  - Checking validity of types...
error - ESLint must be installed in order to run during builds: yarn add --dev eslint
info  - Creating an optimized production build...
info  - Using external babel configuration from /usr/app/.babelrc
info  - Compiled successfully
info  - Collecting page data...

> Build error occurred
Error: You can not use getInitialProps with getStaticProps. To use SSG, please remove your getInitialProps
    at /usr/app/node_modules/next/dist/build/utils.js:595:23
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async Span.traceAsyncFn (/usr/app/node_modules/next/dist/telemetry/trace/trace.js:60:20) {
  type: 'Error'
}
The command '/bin/sh -c npm run build' returned a non-zero code: 1
```

I have `getInitialProps` only in `pages/_document.tsx` and `_error.tsx`, and I can't figure out more importantly why it builds perfectly via local Dockerfile but not AWS...?

### Expected Behavior

The expected behavior is that it builds the app.

### To Reproduce

I'm not sure how give steps to reproduce. Can follow up with more info if needed.

**Comment by ZhelinCheng** [comment]
me too
